### ISSUE TITLE
Issues with Escargot implementation

### ISSUE THREAD

**Issue by Toflar** [contao/contao#3187]
**Affected version(s)**

Contao 4.9 as well as 4.11

**Description**

Issues (or not?):

1. Both, the `BrokenLinkCheckerSubscriber` as well as the `SearchIndexSubscriber` currently do not consider the `disallow:` statements in the `robots.txt` at all.
2. The `BrokenLinkCheckerSubscriber` also ignores `rel="nofollow"` as well as any `robots` `nofollow` information.

This is due to the way, Escargot was designed.
Escargot consists of an arbitrary number of subscribers. Each of them decides individually, whether they want a certain URL to be requested and it does not decide anything itself.
Escargot is a spider library and by no means limited to HTML crawling. Hence, the `robots.txt` handling is completely optional.

I think the fact that the `BrokenLinkCheckerSubscriber` ignores this information is actually correct. Otherwise it could not check links on your site because they were blocked by your `robots.txt` `disallow` definition which sounds strange to me.
However, the `SearchIndexSubscriber` should skip `disallow` definitions in the `robots.txt` too.

If we can agree on that, I'll create a PR.

**Comment by ausi** [comment]
> I think the fact that the `BrokenLinkCheckerSubscriber` ignores this information is actually correct.

Agree.

> However, the `SearchIndexSubscriber` should skip `disallow` definitions in the `robots.txt` too.

Agree, but it should be possible to match a user agent so that one can allow escargot explicitly:

```
User-agent: escargot
Disallow:

User-agent: *
Disallow: /
```

**Comment by Toflar** [comment]
That's already possible, we just ignore it 🙈

**Comment by Toflar** [comment]
Closing in favor of https://github.com/contao/contao/pull/3190 then :)
